client/extensions/woocommerce/state/wc-api/endpoints/products/reducer.js

```diff
@@ -0,0 +1,50 @@
+/** @format */
+
+/**
+ * Internal dependencies
+ */
+import { createReducer } from 'state/utils';
```

Comment:
With an 👁 toward the future wherein this utility could be used in wp-admin, do we want to build upon calypso-centric patterns like this? Just wondering, not implying we shouldn't.

Reply:
That's a fair point. I was really just porting this code over, but more thought should be given to such actions. I'm planning on splitting up some of the work in this and #22537 with proper structure and tests, so I'll keep this in mind when I do.